### DISCUSSION TITLE
API Instagram... alguém ja trabalhou??

### DISCUSSION THREAD

**Discussion by nvrossett** [frontendbr/forum#750] [Perguntas e Respostas]
Alguém ai ja mexeu com API do Instagram?

Queria apenas tirar uma dúvida para avaliar a viabilidade de um projeto que estou orçando.
Eu vi que tem esquema de Webhook's na api deles.

A ideia era ver se tem como eu receber em tempo real quando alguém curti uma perfil no instagram, não só pelo app do cliente mas pela própria pagina do instagram do cliente, para criar interações com ele dentro do sistema.

Não sei se o subscription deles engloba apenas o app sendo desenvolvido ou a pagina toda.
A mesma coisa será feita para fotos, monitorar se a pessoal deu 10 likes em fotos no mês e tal.

A parte de subscription deles é bem escassa de informações
https://www.instagram.com/developer/subscriptions/

Queria saber se recebe eventos gerais do instagram ou somente da aplicação

**Comment by romulo2735** [comment]
Essa API é a mesma que faz Bot para ele?

**Comment by nvrossett** [comment]
Sim acredito que só tenha uma api até aonde sei, a final a função de criar o bot automatizando as ações ou não vai da lógica implementada pelo programador.

Você ja chegou a utilizar?

**Comment by romulo2735** [comment]
Eu tentei usa ela no PHP, meu amigo trabalha com marketing de midias sociais e eu fui atras, tem que ter uma chave para pode utilizar ela sabe. No final ele queria usa a nuvem da Microsoft para colocar no ar, ai ele desistiu, mas deu um trabalho grande, pelo menos para mim que nunca programei para isso.

**Comment by nvrossett** [comment]
Saquei, queria só ter uma noção se o nivel de notificação desse webhook deles é apenas a nivel de interação da aplicação ou de interações gerais da página pois vai ditar todo o trabalho que precisa ser realizado.

Pois o ideal seria se qualquer interação realizada na pagina por dentro da aplicação ou não a minha aplicação pudesse gerar a interatividade com a pessoa por receber a notificação via webhook.

**Comment by romulo2735** [comment]
Hum, mas assim não posso te dizer com mais certeza com funciona porque eu não cheguei tão longe rs.

**Comment by nvrossett** [comment]
heheheheh sem problemas, vamos ver se alguém aqui sabe dizer, mas vlw

**Comment by felipefialho** [comment]
Se ninguém tiver tido essa experiência, não é má ideia perguntar diretamente para os mantenedores da API deles viu...

**Comment by matheusmpessoa** [comment]
A pouco tempo atrás precisei fazer uma integração com o instagram e tive muito trabalho, é sem dúvidas a rede social mais chata de trabalhar.
Minha dica é continuar procurando alguém que já usou essa API, pois a documentação é fraca e com poucos exemplos.

**Comment by brpontes** [comment]
Cara, eu cheguei a começar um mini projeto com essa API mas é realmente muito complicado. Só funciona em Sandbox, não consegui recuperar nenhuma mídia a não ser as minhas 20 últimas imagens... Você consegue também recuperar o número de seguidores, quantidade de comentários das fotos e curtidas, mas é o máximo.

https://github.com/brunodev18/reactjs-instagram

**Comment by dennis90** [comment]
Amigo eu já fiz algumas coisas utilizando o instagram, mas não com a API deles, o único jeito de conseguir algumas informações mais detalhadas é utilizando um "hack" do app mobile deles, tem essa biblioteca desenvolvida em PHP que resolve uma boa parte das necessidades:

https://github.com/mgp25/Instagram-API

**Comment by nvrossett** [comment]
Poo que saco, parecia ser simples, mas então não é tão simples assim...
Realmente a documentação deles é muito fraca =/

@dennis90 do que você ja viu dessa api e as necessidades que demonstrei... acha tranquilo trabalhar com ela? Acha que ela resolve os problemas que levantei no inicio da postagem?

**Comment by dennis90** [comment]
@nvrossett, então eles tem esse exemplo de notificações push:
https://github.com/mgp25/Instagram-API/blob/master/examples/pushReceiver.php

Lá tem exemplo para curtidas, comentários e mensagens diretas, mas não testei ainda porque não tenho um ambiente em PHP aqui para testar...

**Comment by allefbr** [comment]
Sempre tive problema com a API do instagram, até hoje quando preciso fazer alguma coisa uso o PHP, espero que consiga resolver amigo.

**Comment by nvrossett** [comment]
@dennis90  vlw cara

@allefbr  vlw pelo comentário cara, a partir do seu conhecimento com a integração.

"A ideia era ver se tem como eu receber em tempo real quando alguém curti uma perfil no instagram, não só pelo app do cliente mas pela própria pagina do instagram do cliente, para criar interações com ele dentro do sistema.

Não sei se o subscription deles engloba apenas o app sendo desenvolvido ou a pagina toda.
A mesma coisa será feita para fotos, monitorar se a pessoal deu 10 likes em fotos no mês e tal."

O que vc tem a dizer? É viável monitorar os seguidores no instagram em tempo real?

Tipo o cara seguiu (por dentro da aplicação ou não) uma pagina em especifica e ele recebe uma mensagem com um brinde e tal.

A mesma coisa, monitorar se um id de usuário deu like de uma foto para armazenar isso no banco de dados e dar um brinde se no banco mostrar 10 likes no mês e tal.

**Comment by torto** [comment]
Trabalhei recentemente com a api do instagram... Não existe Webhook's de fato na api deles...
O subscription (quer era para ser algo decente) envia apenas as imagens novas add pelos usuários que fizeram login no app q vc criou... ele envia id do usuário e id da media e só!
Para saber quais os comentários, vc vai ter q bater no rest com esse id da media e com o token do usuário e ele vai trazer os últimos 150 comentários apenas.. ou seja vc vai ter q fazer algo q fica batendo na api de tempos em tempos para coletar esses novos comentários...
Não sei se vc consegue captar quais as fotos o usuário deu curtir...
De fato é uma api simples.. login para ela eh a coisa mais simples.. mas ta na simplicidade que torna ela uma merda... vc não tem quase nada de info...

**Reply by felquis** [reply]
@torto existe webhook em 2022?

**Comment by nvrossett** [comment]
@torto pooo que merda, puta sistema com uma equipe por trás e uma api dessas é lamentável...
Vlw mesmo cara, seu comentário ajudou muito

**Comment by kelvysmoura** [comment]
> Amigo eu já fiz algumas coisas utilizando o instagram, mas não com a API deles, o único jeito de conseguir algumas informações mais detalhadas é utilizando um "hack" do app mobile deles, tem essa biblioteca desenvolvida em PHP que resolve uma boa parte das necessidades:
> 
> https://github.com/mgp25/Instagram-API

Eu encontrei esse repositorio recentemente e realmente é muito bom, mas queria saber quais as chance de parar de funcionar com as novas mudanças de 2018 na api do instagram. 

Alguém que já usou ou usa sabe dizer se realmente da pra confiar e implementar em projetos de clientes?

**Comment by RubenCollins** [comment]
Has anyone wondered how [baixar stories do instagram privado](https://instazoomhd.com/pt/) to download private photos, it is the solution to solve what you are wondering.
